driver/src/main/java/org/neo4j/driver/v1/GraphDatabase.java

```diff
@@ -151,97 +122,9 @@ public static Driver driver( String uri, AuthToken authToken, Config config )
-                new SocketConnectionPool( connectionSettings, securityPlan, poolSettings, config.logging() );
-        switch ( scheme.toLowerCase() )
-        {
-        case "bolt":
-            return new DirectDriver( address, connectionPool, securityPlan, config.logging() );
-        case "bolt+routing":
-            return new RoutingDriver(
-                    config.routingSettings(),
-                    address,
-                    connectionPool,
-                    securityPlan,
-                    Clock.SYSTEM,
-                    config.logging() );
-        default:
-            throw new ClientException( format( "Unsupported URI scheme: %s", scheme ) );
-        }
-    }
-
-    /*
-     * Establish a complete SecurityPlan based on the details provided for
-     * driver construction.
-     */
-    private static SecurityPlan createSecurityPlan( BoltServerAddress address, Config config )
-            throws GeneralSecurityException, IOException
-    {
-        Config.EncryptionLevel encryptionLevel = config.encryptionLevel();
-        boolean requiresEncryption = encryptionLevel.equals( REQUIRED );
-
-        if ( requiresEncryption )
-        {
-            Logger logger = config.logging().getLog( "session" );
-            switch ( config.trustStrategy().strategy() )
-            {
-
-            // DEPRECATED CASES //
-            case TRUST_ON_FIRST_USE:
-                logger.warn(
-                        "Option `TRUST_ON_FIRST_USE` has been deprecated and will be removed in a future " +
-                        "version of the driver. Please switch to use `TRUST_ALL_CERTIFICATES` instead." );
-                return SecurityPlan.forTrustOnFirstUse( config.trustStrategy().certFile(), address, logger );
-            case TRUST_SIGNED_CERTIFICATES:
-                logger.warn(
-                        "Option `TRUST_SIGNED_CERTIFICATE` has been deprecated and will be removed in a future " +
-                        "version of the driver. Please switch to use `TRUST_CUSTOM_CA_SIGNED_CERTIFICATES` instead." );
-                // intentional fallthrough
-            // END OF DEPRECATED CASES //
+        config = config == null ? Config.defaultConfig() : config;
 
-            case TRUST_CUSTOM_CA_SIGNED_CERTIFICATES:
-                return SecurityPlan.forCustomCASignedCertificates( config.trustStrategy().certFile() );
-            case TRUST_SYSTEM_CA_SIGNED_CERTIFICATES:
-                return SecurityPlan.forSystemCASignedCertificates();
-            case TRUST_ALL_CERTIFICATES:
-                return SecurityPlan.forAllCertificates();
-            default:
-                throw new ClientException(
-                        "Unknown TLS authentication strategy: " + config.trustStrategy().strategy().name() );
-            }
-        }
-        else
-        {
-            return insecure();
-        }
+        return new DriverFactory().newInstance( uri, authToken, config.routingSettings(), config );
```

Comment:
I don't know, how about a static factory instance or a static newInstance method? Feels strange to create a stateless Factory object every time, but maybe you have a reason...